std/algorithm/mutation.d

```diff
@@ -1923,6 +1924,71 @@ if (s == SwapStrategy.stable
     return result;
 }
 
+/// ditto
+// SwapStrategy s is ignored here, it's just always stable,
+// unstable doesn't really work with variable length encoding
+auto remove
+(SwapStrategy s = SwapStrategy.stable, Range, Offset...)
+(Range range, Offset offsets)
+if (isNarrowString!Range
+    && isMutable!(typeof(range[0]))
+    && Offset.length >= 1)
```

Comment:
Please use `static if` internally - the user doesn't care about the implementation details.

Reply:
You mean combine with the non-string overloads? I imagine one overload that takes offsets and one that takes a predicate.

Reply:
> You mean combine with the non-string overloads?

Yep, the rest is implementation detail.

>  I imagine one overload that takes offsets and one that takes a predicate.

Yup, but these should use `static if` internally.

Reply:
Given `A(T) if (conditionA!T)` and `B(T) if (conditionB!T)` to merge in to one template, is there a strong preference to do

`A(T) if (conditionA!T || conditionB!T)
{
    static if (conditionA!T) { ... }
    else { static assert(conditionB!T); ... }
}`

or

`A(T)
{
    static if (conditionA!T) { ... }
    else static if (conditionB!T) { ... }
    else static assert(false, "Can't do A with ` ~ T.stringof);
}`

in phobos.

Reply:
I think there exists no strong guideline on this yet.
However, in the past we went more often with `static assert(0)`due to the better error messages and because it avoids the bulky constraints.
So ideally it's something like:

```d
A(T)
if (simplifiedCommonCondition)
{
    static if (conditionA!T) { ... }
    else static if (conditionB!T) { ... }
    else static assert(false, "Can't do A with ` ~ T.stringof);
}
```

Reply:
I've made an attempt at this.